server/apps/main/helpers.py

```diff
@@ -31,6 +31,11 @@ def model_to_form(model, disable_moderator=False):
     model_dict = model_to_dict(model)
     model_dict["moderation_prior"] = model_dict["moderation_status"]
 
+    # Copy the latest moderation reply to the form
+    if (model.experience_id):
```

Comment:
I'm not 100% sure I understand why this is necessary. Is it to handle the case where someone force submits a form (i.e. client-side JS doesn't prevent non-valid form submitted) and we need to restore the form?

Reply:
This intention here was more to do with moderator workflow, building on the issue flagged by @mastoffel that the moderator can't 'edit' previous feedback. This change ensures that the form is pre-populated with the moderation replies from the _previous_ moderation round. This seems relevant especially for the 'in review' state, where the moderator may be returning to a half-completed review, but potentially also useful in the other cases as well.

But this is a change in behaviour, so up for discussion of course!

I had to check about the case you describe, where the user is bounced back after a failed validation, but in that case django seems to retain the old values automatically (so this change could be removed without affecting that).

Reply:
Ah of course, that makes total sense. I had forgotten about needing to re-populate the fields with existing bits! 👍